src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1821,6 +1823,66 @@ export class ResultsViewPageStore {
         }
     });
 
+    readonly expressionProfiles = remoteData({
+        await:()=>[
+            this.molecularProfilesInStudies
+        ],
+        invoke:()=>{
+            return Promise.resolve(this.molecularProfilesInStudies.result.filter(
+                (profile:MolecularProfile)=>isExpressionProfile(profile.molecularProfileId, this.expressionTabseqVersion)
+            ));
+        }
+    });
+
+    readonly rnaSeqMolecularData = remoteData<{[hugoGeneSymbol:string]:NumericGeneMolecularData[][]}>({
+       await:()=>[
+           this.expressionProfiles,
+           this.genes,
+           this.geneMolecularDataCache
+       ],
+       invoke: async ()=>{
+
+           const rnaSeqProfiles = this.expressionProfiles.result!;
+
+           const queries = _.flatMap(this.genes.result,(gene:Gene)=>{
+               return rnaSeqProfiles.map((profile:MolecularProfile)=> {
+                   return ({
+                       entrezGeneId: gene.entrezGeneId,
+                       molecularProfileId: profile.molecularProfileId,
+                       hugoGeneSymbol:gene.hugoGeneSymbol
+                   })
+               });
+           });
+
+           const data = await this.geneMolecularDataCache.result!.getPromise(queries,true);
+
+           // group cache objects by entrez geneId
+           //TODO: fix typing of cachItem
```

Comment:
@adamabeshouse  for some reason, i couldn't get tying of cachItem to work

Reply:
the type of `cacheItem` is `CacheData<NumericGeneMolecularData[], { entrezGeneId:number, molecularProfileId:string; }>`